From: Jan Grulich <jgrulich@redhat.com>
Date: Sat, 15 Jun 2024 11:27:00 +0000
Subject: Bug 1902618 - WebRTC backport: PipeWire video capture: mmap()
 PipeWire buffers with MAP_SHARED r=pehrsons,webrtc-reviewers

This is a simple backport of an WebRTC upstream change.

Upstream commit: 025d69b4d0df05f54d7bfcb3a4a2b35b48b83d82

Differential Revision: https://phabricator.services.mozilla.com/D213752
Mercurial Revision: https://hg.mozilla.org/mozilla-central/rev/09f0b8ff3d06a68baefa88ed598d2125dfe9d417
---
 modules/video_capture/linux/video_capture_pipewire.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/video_capture/linux/video_capture_pipewire.cc b/modules/video_capture/linux/video_capture_pipewire.cc
index d7cd50df66..9ff4fdb9b1 100644
--- a/modules/video_capture/linux/video_capture_pipewire.cc
+++ b/modules/video_capture/linux/video_capture_pipewire.cc
@@ -418,7 +418,7 @@ void VideoCaptureModulePipeWire::ProcessBuffers() {
       ScopedBuf frame;
       frame.initialize(
           static_cast<uint8_t*>(
-              mmap(nullptr, spaBuffer->datas[0].maxsize, PROT_READ, MAP_PRIVATE,
+              mmap(nullptr, spaBuffer->datas[0].maxsize, PROT_READ, MAP_SHARED,
                    spaBuffer->datas[0].fd, spaBuffer->datas[0].mapoffset)),
           spaBuffer->datas[0].maxsize, spaBuffer->datas[0].fd,
           spaBuffer->datas[0].type == SPA_DATA_DmaBuf);
